azure/azure-iot-hub-device-simulator/src/main/java/org/eclipse/examples/azure/DittoAzureIoTHubClient.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.examples.azure;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import javax.annotation.Nullable;
+
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.json.JsonPointer;
+import org.eclipse.ditto.model.base.common.HttpStatusCode;
+import org.eclipse.ditto.model.base.headers.DittoHeaders;
+import org.eclipse.ditto.model.things.Feature;
+import org.eclipse.ditto.model.things.ThingId;
+import org.eclipse.ditto.protocoladapter.Payload;
+import org.eclipse.ditto.protocoladapter.ProtocolFactory;
+import org.eclipse.ditto.protocoladapter.TopicPath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.microsoft.azure.sdk.iot.device.DeviceClient;
+import com.microsoft.azure.sdk.iot.device.DeviceTwin.DeviceMethodCallback;
+import com.microsoft.azure.sdk.iot.device.DeviceTwin.DeviceMethodData;
+import com.microsoft.azure.sdk.iot.device.IotHubConnectionStatusChangeCallback;
+import com.microsoft.azure.sdk.iot.device.IotHubConnectionStatusChangeReason;
+import com.microsoft.azure.sdk.iot.device.IotHubEventCallback;
+import com.microsoft.azure.sdk.iot.device.IotHubMessageResult;
+import com.microsoft.azure.sdk.iot.device.IotHubStatusCode;
+import com.microsoft.azure.sdk.iot.device.Message;
+import com.microsoft.azure.sdk.iot.device.ProxySettings;
+import com.microsoft.azure.sdk.iot.device.transport.IotHubConnectionStatus;
+
+public final class DittoAzureIoTHubClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DittoAzureIoTHubClient.class.getSimpleName());
+
+    private final DeviceClient client;
+    private final ConfigProperties properties;
+
+    DittoAzureIoTHubClient() throws URISyntaxException {
+        properties = ConfigProperties.getInstance();
```

Comment:
I suggest to provide properties and client as arguments to the constructor. Setting the proxy settings could remain in this class. If it had a static factory method then this would be the place to call `setProxyToDeviceClientIfEnabled` and `registerConnectionStatusChangeCallback` on the provided client. 

Doing so only the main method had to deal with exceptions and the two aspects of creating and using the client would be separated more strictly.